### PR TITLE
MERC-5889:  SBRV Tests for VerifierProxy

### PR DESCRIPTION
**Pull request by dav009**
[MERC-5889](https://smartcontract-it.atlassian.net/browse/MERC-5889)

### Resolves Dependencies
- https://github.com/smartcontractkit/chainlink/pull/13813

## What ?

- Added Tests for `DestinationVerifierProxy`
- Added a couple changes to contract to make tests pass 
   
  

[MERC-5889]: https://smartcontract-it.atlassian.net/browse/MERC-5889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol [hunk 1]
```diff
@@ -37,14 +37,16 @@ contract DestinationVerifierProxy is IDestinationVerifierProxy, ConfirmedOwner,
     bytes calldata payload,
     bytes calldata parameterPayload
   ) external payable returns (bytes memory) {
+    if(address(s_verifier) == address(0)) revert ZeroAddress();
```
Comment (Fletch153) [inline contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol]: We don't want to check this as reading and writing from contract storage is expensive. For reference:

Writing to storage where existing element is 0: 20,000 gas
Writing to storage where existing element is non 0: 5,000 gas
Reading from cold storage (1st read) : 2,000
Reading from warm storage (2nd+ read) : 100
Allocating a uint256 in memory: <30 ( memory allocation cost grows quadratically)

Given we'll always smoke test before we release the contract and a nil or bad verifier cant be set through the setter, adding this check adds additional cost to the user (2% more for verification)

File: contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol [hunk 2]
```diff
@@ -53,10 +53,11 @@ contract DestinationVerifierProxy is IDestinationVerifierProxy, ConfirmedOwner,
     if(verifierAddress == address(0)) revert ZeroAddress();
```
Comment (Fletch153) [inline contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol]: We should remove this line also

**Review by Fletch153 — APPROVED** [review]

File: contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol (hunk 2 above)
Comment (Fletch153) [reply contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol]: ^

Comment (dav009) [reply contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol]: 🙏🏼  happy to remove this, but this check is part of setVerifier, not part of verifyBulk. 
does it still affect gas?

Comment (Fletch153) [reply contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol]: it does - it is the same logic, we're reading from storage  when we know verifierAddress will never be 0 once we've initialised and smoke tested

Comment (Fletch153) [inline contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifierProxy.sol]:
```suggestion
```

**Review by Fletch153 — APPROVED** [review]

**Pull request merged**